mqtt.go

```diff
@@ -383,29 +383,32 @@ func (mc *mqttConn) sendEvent(e *DeviceEvent) error {
 		Payload: payload,
 	}
 
-	for count := uint(1); count <= maxDeviceEventAttempts; count++ {
-		mc.outPacket <- p
+	mc.outPacket <- p
 
-		if e.qos == QOSAtMostOnce {
-			return nil
-		}
+	logInfo("[MQTT] event delivery for packet ID %d", p.PacketID)
 
-		logInfo("[MQTT] event delivery attempt #%d for packet ID %d", count, p.PacketID)
-		logInfo("[MQTT] waiting for PUBACK for packet ID %d", p.PacketID)
+	if e.qos == QOSAtMostOnce {
+		return nil
+	}
 
-		select {
-		case <-time.After(deviceEventRetryInterval):
-			logError("[MQTT] did not receive PUBACK for packet ID %d within %v", p.PacketID, deviceEventRetryInterval)
+	logInfo("[MQTT] waiting for PUBACK for packet ID %d", p.PacketID)
 
-		case ack := <-mc.puback:
-			if ack.PacketID == p.PacketID {
-				logInfo("[MQTT] received PUBACK for packet ID %d", ack.PacketID)
-				return nil
-			}
-			// TBD: Something is really wrong if packet ID does not match. What to do?
+	select {
+	case <-time.After(deviceEventRetryInterval):
+		msg := fmt.Sprintf("did not receive PUBACK of event delivery for packet ID %d", p.PacketID)
+		logError("[MQTT] %s", msg)
+		mc.reportError(fmt.Errorf("event delivery timeout: %s", msg))
+
+	case ack := <-mc.puback:
+		if ack.PacketID == p.PacketID {
+			logInfo("[MQTT] received PUBACK for packet ID %d", ack.PacketID)
+			return nil
 		}
 
-		p.Dup = true
+		msg := fmt.Sprintf("sendEvent() received packet that does not match the Ack packet ID (%d) and the expected packet ID (%d).",
+			ack.PacketID, p.PacketID)
+		logInfo("[MQTT] %s", msg)
+		mc.reportError(fmt.Errorf("event delivery unmatched Ack Packet: %s", msg))
```

Comment:
In the same way, call reportError for reconnect if it received another packet id. Because, MQTT connection is not stable in this situation.